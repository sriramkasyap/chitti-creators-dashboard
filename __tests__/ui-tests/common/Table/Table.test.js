import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import Table from "../../../../src/components/common/Table/Table";

const data = [
  {
    email: "sara.andersen@example.com",
    name: "Ms. Sara Andersen",
    subscriptionType: "Free",
  },
  {
    email: "edita.vestering@example.com",
    name: "Miss. Edita Vestering",
    subscriptionType: "Free",
  },
  {
    email: "roberto.vega@example.com",
    name: "Mr. Roberto Vega",
    subscriptionType: "Paid",
  },
  {
    email: "rudi.droste@example.com",
    name: "Mr. Rudi Droste",
    subscriptionType: "Free",
  },
  {
    email: "carolina.lima@example.com",
    name: "Mrs. Carolina Lima",
    subscriptionType: "Paid",
  },
  {
    email: "emre.asikoglu@example.com",
    name: "Mr. Emre Asikoglu",
    subscriptionType: "Free",
  },
  {
    email: "kent.brewer@example.com",
    name: "Mr. Kent Brewer",
    subscriptionType: "Free",
  },
  {
    email: "evan.carlson@example.com",
    name: "Mr. Evan Carlson",
    subscriptionType: "Free",
  },
  {
    email: "friedrich-karl.brand@example.com",
    name: "Mr. Friedrich-Karl Brand",
    subscriptionType: "Paid",
  },
  {
    email: "valentin.ortega@example.com",
    name: "Mr. Valentin Ortega",
    subscriptionType: "Free",
  },
];

const columns = [
  {
    Header: "Name",
    accessor: "name",
  },
  {
    Header: "Email",
    accessor: "email",
  },
  {
    Header: "Subscription Type",
    accessor: "subscriptionType",
  },
];

describe("Alert initial render", () => {
  it("should match the snapshot", () => {
    const container = render(
      <Table data={data} columns={columns} hiddenColumns={["name"]} size="md" />
    );

    expect(container).toMatchSnapshot();
  });
});
