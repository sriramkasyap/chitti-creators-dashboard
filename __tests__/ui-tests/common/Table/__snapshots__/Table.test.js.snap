// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert initial render should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <table
        class="chakra-table css-0"
        role="table"
      >
        <thead
          class="css-0"
        >
          <tr
            class="css-0"
            role="row"
          >
            <th
              class="css-mvn64f"
              colspan="1"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="css-u4p24i"
              >
                <p
                  class="chakra-text css-0"
                >
                  Email
                </p>
                <span
                  class="css-rkk3mp"
                />
              </div>
            </th>
            <th
              class="css-mvn64f"
              colspan="1"
              role="columnheader"
              style="cursor: pointer;"
              title="Toggle SortBy"
            >
              <div
                class="css-u4p24i"
              >
                <p
                  class="chakra-text css-0"
                >
                  Subscription Type
                </p>
                <span
                  class="css-rkk3mp"
                />
              </div>
            </th>
          </tr>
        </thead>
        <tbody
          class="css-0"
          role="rowgroup"
        >
          <tr
            class="css-0"
            role="row"
          >
            <td
              class="css-j251j9"
              role="cell"
            >
              sara.andersen@example.com
            </td>
            <td
              class="css-j251j9"
              role="cell"
            >
              Free
            </td>
          </tr>
          <tr
            class="css-0"
            role="row"
          >
            <td
              class="css-j251j9"
              role="cell"
            >
              edita.vestering@example.com
            </td>
            <td
              class="css-j251j9"
              role="cell"
            >
              Free
            </td>
          </tr>
          <tr
            class="css-0"
            role="row"
          >
            <td
              class="css-j251j9"
              role="cell"
            >
              roberto.vega@example.com
            </td>
            <td
              class="css-j251j9"
              role="cell"
            >
              Paid
            </td>
          </tr>
          <tr
            class="css-0"
            role="row"
          >
            <td
              class="css-j251j9"
              role="cell"
            >
              rudi.droste@example.com
            </td>
            <td
              class="css-j251j9"
              role="cell"
            >
              Free
            </td>
          </tr>
          <tr
            class="css-0"
            role="row"
          >
            <td
              class="css-j251j9"
              role="cell"
            >
              carolina.lima@example.com
            </td>
            <td
              class="css-j251j9"
              role="cell"
            >
              Paid
            </td>
          </tr>
          <tr
            class="css-0"
            role="row"
          >
            <td
              class="css-j251j9"
              role="cell"
            >
              emre.asikoglu@example.com
            </td>
            <td
              class="css-j251j9"
              role="cell"
            >
              Free
            </td>
          </tr>
          <tr
            class="css-0"
            role="row"
          >
            <td
              class="css-j251j9"
              role="cell"
            >
              kent.brewer@example.com
            </td>
            <td
              class="css-j251j9"
              role="cell"
            >
              Free
            </td>
          </tr>
          <tr
            class="css-0"
            role="row"
          >
            <td
              class="css-j251j9"
              role="cell"
            >
              evan.carlson@example.com
            </td>
            <td
              class="css-j251j9"
              role="cell"
            >
              Free
            </td>
          </tr>
          <tr
            class="css-0"
            role="row"
          >
            <td
              class="css-j251j9"
              role="cell"
            >
              friedrich-karl.brand@example.com
            </td>
            <td
              class="css-j251j9"
              role="cell"
            >
              Paid
            </td>
          </tr>
          <tr
            class="css-0"
            role="row"
          >
            <td
              class="css-j251j9"
              role="cell"
            >
              valentin.ortega@example.com
            </td>
            <td
              class="css-j251j9"
              role="cell"
            >
              Free
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="chakra-table css-0"
      role="table"
    >
      <thead
        class="css-0"
      >
        <tr
          class="css-0"
          role="row"
        >
          <th
            class="css-mvn64f"
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="css-u4p24i"
            >
              <p
                class="chakra-text css-0"
              >
                Email
              </p>
              <span
                class="css-rkk3mp"
              />
            </div>
          </th>
          <th
            class="css-mvn64f"
            colspan="1"
            role="columnheader"
            style="cursor: pointer;"
            title="Toggle SortBy"
          >
            <div
              class="css-u4p24i"
            >
              <p
                class="chakra-text css-0"
              >
                Subscription Type
              </p>
              <span
                class="css-rkk3mp"
              />
            </div>
          </th>
        </tr>
      </thead>
      <tbody
        class="css-0"
        role="rowgroup"
      >
        <tr
          class="css-0"
          role="row"
        >
          <td
            class="css-j251j9"
            role="cell"
          >
            sara.andersen@example.com
          </td>
          <td
            class="css-j251j9"
            role="cell"
          >
            Free
          </td>
        </tr>
        <tr
          class="css-0"
          role="row"
        >
          <td
            class="css-j251j9"
            role="cell"
          >
            edita.vestering@example.com
          </td>
          <td
            class="css-j251j9"
            role="cell"
          >
            Free
          </td>
        </tr>
        <tr
          class="css-0"
          role="row"
        >
          <td
            class="css-j251j9"
            role="cell"
          >
            roberto.vega@example.com
          </td>
          <td
            class="css-j251j9"
            role="cell"
          >
            Paid
          </td>
        </tr>
        <tr
          class="css-0"
          role="row"
        >
          <td
            class="css-j251j9"
            role="cell"
          >
            rudi.droste@example.com
          </td>
          <td
            class="css-j251j9"
            role="cell"
          >
            Free
          </td>
        </tr>
        <tr
          class="css-0"
          role="row"
        >
          <td
            class="css-j251j9"
            role="cell"
          >
            carolina.lima@example.com
          </td>
          <td
            class="css-j251j9"
            role="cell"
          >
            Paid
          </td>
        </tr>
        <tr
          class="css-0"
          role="row"
        >
          <td
            class="css-j251j9"
            role="cell"
          >
            emre.asikoglu@example.com
          </td>
          <td
            class="css-j251j9"
            role="cell"
          >
            Free
          </td>
        </tr>
        <tr
          class="css-0"
          role="row"
        >
          <td
            class="css-j251j9"
            role="cell"
          >
            kent.brewer@example.com
          </td>
          <td
            class="css-j251j9"
            role="cell"
          >
            Free
          </td>
        </tr>
        <tr
          class="css-0"
          role="row"
        >
          <td
            class="css-j251j9"
            role="cell"
          >
            evan.carlson@example.com
          </td>
          <td
            class="css-j251j9"
            role="cell"
          >
            Free
          </td>
        </tr>
        <tr
          class="css-0"
          role="row"
        >
          <td
            class="css-j251j9"
            role="cell"
          >
            friedrich-karl.brand@example.com
          </td>
          <td
            class="css-j251j9"
            role="cell"
          >
            Paid
          </td>
        </tr>
        <tr
          class="css-0"
          role="row"
        >
          <td
            class="css-j251j9"
            role="cell"
          >
            valentin.ortega@example.com
          </td>
          <td
            class="css-j251j9"
            role="cell"
          >
            Free
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
